src/Microsoft.AspNet.Mvc.Core/Formatters/FormatFilter.cs

```diff
@@ -65,13 +61,15 @@ public void OnResourceExecuting(ResourceExecutingContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            if (!IsActive)
+            var format = GetFormat(context);
+            if (format == null)
             {
                 // no format specified by user, so the filter is muted
                 return;
             }
 
-            if (ContentType == null)
+            var contentType = _options.FormatterMappings.GetMediaTypeMappingForFormat(format);
```

Comment:
Note the change here `ContentType` -> `contentType`, which is why the lambda below might make you do a double-take.